core/types/receipt.go

```diff
@@ -182,8 +200,8 @@ func (r Receipt) EncodeRLP(w io.Writer) error {
 	buf := new(bytes.Buffer)
 	buf.WriteByte(r.Type)
 	if r.Type == DepositTxType {
-		withNonce := &depositReceiptRlp{r.statusEncoding(), r.CumulativeGasUsed, r.Bloom, r.Logs, r.DepositNonce}
-		if err := rlp.Encode(buf, withNonce); err != nil {
+		withNonceAndReceiptVersion := &depositReceiptRlp{r.statusEncoding(), r.CumulativeGasUsed, r.Bloom, r.Logs, r.DepositNonce, r.DepositReceiptVersion}
```

Comment:
shouldn't we check if deposit receipt version is nil?

Reply:
nil check is done using [struct tags](https://pkg.go.dev/github.com/CortexFoundation/CortexTheseus/rlp#hdr-Struct_Tags): ``rlp:"optional"`.

https://github.com/testinprod-io/op-erigon/blob/3e319d2f63d7bdae176263451208644ee7b7e559/core/types/receipt.go#L125-L137

Therefore if `DepositReceiptVersion == nil`, RLP encoder will omit the field